MSAL/test/integration/native_auth/end_to_end/otp_code_retriever/MSALNativeAuthEmailCodeRetriever.swift

```diff
@@ -76,7 +84,19 @@ class MSALNativeAuthEmailCodeRetriever: XCTestCase {
             }
             if dataDictionary.count > 0 {
                 dataDictionary.sort(by: {($0["id"] as? Int ?? 0) > ($1["id"] as? Int ?? 0)})
-                return dataDictionary.first?["id"] as? Int
+                if let emailDateString = dataDictionary.first?["date"] as? String,
+                   let emailDate = dateFormatter.date(from: emailDateString) {
+                    let currentDate = Date()
+                    // Email should be newer than 5 seconds otherwise it could be from previous test
+                    // This retry will help with the delay in receiving the emails
+                    if currentDate.timeIntervalSince1970 - emailDate.timeIntervalSince1970  < secondsToWaitForEmail {
+                        print ("Email is for current test, last receive date: \(emailDate) current date: \(currentDate)")
+                        return dataDictionary.first?["id"] as? Int
+                    } else {
+                        print ("Email is from previous tests, last receive date: \(emailDate) current date: \(currentDate)")
+                    }
+                }
+                return await retrieveLastMessage(local: local, domain: domain, retryCounter: retryCounter - 1)
```

Comment:
you could remove the line 99 and remove the else at line 100. So, we can keep printing the error message at line 103 at last retry and call the "retrieveLastMessage" from one single place

Reply:
Indeed, done